torch/csrc/jit/codegen/cuda/inline_propagator.cpp

```diff
@@ -373,23 +385,71 @@ void InlinePropagator::propagateTvSibling(TensorView* from, TensorView* to) {
   recordReplayedPos(to, from_pos);
 }
 
+namespace {
+
 // Try to find the aligned position on consumer's domain corresponding to the
-// compute at position of producer domain.
-void MaxProducerPosUpdater::handle(TensorView* consumer) {
+//  compute at position of producer domain. Used in computeAt pass only. No
+//  checking on actual producer-consumer relationship.
+unsigned int getConsumerPosAlignedToProducerCA(
+    TensorView* consumer,
+    TensorView* producer) {
+  // Locate consumer's position that aligns with
+  //  the producer's new compute at axis. We need broadcast axes forwarded so we
+  //  need to replay PasC as CasP will not forward braodcast dims. For example
+  //  if we have:
+  // T2[ iS22{( 3 * 1 )} ] ca_pos( 1 ) = broadcast( T1[ iS1{3} ] ca_pos( 1 )
+  // produce_pos( 1) ) CasP will have the mapping iS1{3} -> iS2{3} and PasC will
+  // have the mapping iS22{( 3 * 1 )} <- iS1{3} We need the latter. Refer to
+  // NVFuserTest.FusionComplexBCast1_CUDA
+
+  auto c2p_map =
+      BestEffortReplay::replayPasC(
+          producer,
+          consumer,
+          -1,
+          // Compute at root domain may not be valid here, as all
+          // producers don't have to be able to map into consumer at
+          // max producer position. Since computeAt should be valid
+          // and this mechanism is only intended to lower produce
+          // position of consumer, we can simply use the pairwise map.
+          PairwiseRootDomainMap(producer, consumer))
+          .getReplay();
+
+  // Find the innermost position of consumer that has
+  //  been mapped within the producer ca axis.
   unsigned int consumer_pos = consumer->nDims();
   while (consumer_pos > 0) {
-    for (auto producer : ir_utils::producerTvsOf(consumer)) {
-      auto producer_pos = TransformReplay::getMatchedLeafPosWithoutReplayPasC(
-          producer, consumer, consumer_pos);
-      if (producer_pos >= 0 &&
-          producer_pos <= producer->getComputeAtPosition()) {
-        goto finished;
-      }
```

Comment:
I reverted this part back to the copy-pasted code from the previously computeAt. Looks like I was doing something wrong in my new code, and after the `getMatchedLeafPosWithoutReplayPasC` change, this starts to fail.